superset-frontend/src/SqlLab/reducers/getInitialState.js

```diff
@@ -19,6 +19,14 @@
 import { t } from '@superset-ui/core';
 import getToastsFromPyFlashMessages from 'src/components/MessageToasts/getToastsFromPyFlashMessages';
 
+export function dedupeTabHistory(tabHistory) {
```

Comment:
I think you can maybe just skip appending if the last id is the same as current id? Removing duplicates are not necessary when things are working as expected. If a users's localStorage already has bad states because of this, we can just ask them to clear the local storage or close the tabs manually themelves.